TYPO3.Flow/Classes/TYPO3/Flow/Property/TypeConverter/ObjectConverter.php

```diff
@@ -138,9 +141,18 @@ public function getTypeOfChildProperty($targetType, $propertyName, PropertyMappi
         } else {
             $targetPropertyNames = $this->reflectionService->getClassPropertyNames($targetType);
             if (in_array($propertyName, $targetPropertyNames)) {
-                $values = $this->reflectionService->getPropertyTagValues($targetType, $propertyName, 'var');
-                if (count($values) > 0) {
-                    return current($values);
+                $varTagValues = $this->reflectionService->getPropertyTagValues($targetType, $propertyName, 'var');
+                if (count($varTagValues) > 0) {
+                    // This ensures that FQCNs are returned without leading backslashes. Otherwise, something like @var \DateTime
+                    // would not find a property mapper. It is needed because the ObjectConverter doesn't use class schemata,
+                    // but reads the annotations directly.
+                    $declaredType = strtok(trim(current($varTagValues), " \n\t"), " \n\t");
+                    try {
+                        $parsedType = TypeHandling::parseType($declaredType);
+                    } catch (InvalidTypeException $exception) {
+                        throw new \InvalidArgumentException(sprintf($exception->getMessage(), 'class "' . $targetType . '" for property "' . $propertyName . '"'), 1467699674);
+                    }
+                    return $parsedType['type'];
```

Comment:
Last nitpick: This should return the full type including `elementType` if it exists. Else collections/arrays of objects/entities would not be converted correctly.

`return $parsedType['type'] . ($parsedType['elementType'] !== null ? '<' . $parsedType['elementType'] . '>' : '');`